ocaml/typing/types.mli

```diff
@@ -219,14 +219,15 @@ and abbrev_memo =
 
 (** Jkinds classify types. *)
 (* CR layouts v2.8: Say more here. *)
+and type_jkind = type_expr Jkind_types.Type.t
 and jkind = type_expr Jkind_types.t
 
-(* jkind depends on types defined in this file, but Jkind.equal is required
+(* jkind depends on types defined in this file, but Jkind.Type.equal is required
    here. When jkind.ml is loaded, it calls set_jkind_equal to fill a ref to the
    function. *)
 (** INTERNAL USE ONLY
-    jkind.ml should call this with the definition of Jkind.equal *)
-val set_jkind_equal : (jkind -> jkind -> bool) -> unit
+    jkind.ml should call this with the definition of Jkind.Type.equal *)
+val set_jkind_equal : (type_jkind -> type_jkind -> bool) -> unit
```

Comment:
Does this work over all jkinds or just type_jkinds? The name of the function and its type disagree.

Reply:
In this case it only gets used by  `Types.equal_record_representation` and `Types.equal_variant_representation`, so it should be for the better if this gets renamed to `set_type_jkind_equal` with same types.